src/components/common/IdContainer.ts

```diff
@@ -0,0 +1,32 @@
+import _ from 'lodash/fp'
+import { useMemo, useState } from 'react'
+
+/**
+ * DEPRECATED - should switch to useUniqueIdFn pattern
+ */
+export const IdContainer = ({ children }) => {
+  const [id] = useState(() => _.uniqueId('element-'))
+  return children(id)
+}
+
+export type UniqueIdFn = (prefix: string) => string
+
+/**
+ * returns a helper function that accepts a prefix to append to an id that is unique
+ * for the consuming component.
+ * @example
+ * // inside a react component:
+ * const unique = useUniqueIdFn()
+ * //...
+ * div({ id: unique('my-thing-one') }, [...],
+ * div({ id: unique('my-thing-two') }, [...]
```

Comment:
In general one only needs an id for linking together labels/inputs (which actually could be more easily done simply by nesting the input inside of the label component) for linking for things like aria-describedby. Can you make the example show a pattern of how it would actually be used?

Reply:
Also, is the prefix optional? I don't really understand why a prefix is needed at all, so there should be a sensible default value.

Reply:
right now it is not optional... see my other comment.  But I am happy for this conversation :)